src/wallet/wallet.cpp

```diff
@@ -100,27 +100,29 @@ CPubKey CWallet::GenerateNewKey(bool internal)
 
     // use HD key derivation if HD was enabled during wallet creation
     if (IsHDEnabled()) {
-        DeriveNewChildKey(metadata, secret, (CanSupportFeature(FEATURE_HD_SPLIT) ? internal : false));
+        DeriveNewChildKey(walletdb, metadata, secret, (CanSupportFeature(FEATURE_HD_SPLIT) ? internal : false));
     } else {
         secret.MakeNewKey(fCompressed);
     }
 
     // Compressed public keys were introduced in version 0.6.0
-    if (fCompressed)
+    if (fCompressed) {
```

Comment:
Yes, this seems to be a "damned if you do, damned if you don't" thing.
- If you deliberately make style fixes to code you touch, you get comments that it's unrelated
- If you deliberately don't make style fixes to code you touch, you get questions why you didn't update the code for the style guidelines
I think both are valid, unless it makes review-ability in which case it could still be done in a separate commit in the same PR.